src/joy.c

```diff
@@ -845,6 +856,39 @@ int main (int argc, char **argv) {
 
         }
 
+#ifndef _WIN32
+        /*
+         * drop privileges once pcap handle exists
+         */
+        pw = getpwnam(config.username);
+        if (pw) {
+            if (initgroups(pw->pw_name, pw->pw_gid) != 0 ||
+                setgid(pw->pw_gid) != 0 || setuid(pw->pw_uid) != 0) {
+                fprintf(stderr, "error: could not change to '%.32s' uid=%lu gid=%lu: %s\n",
```

Comment:
How does the .32 change the format of string here? Can you get away just %s?

Reply:
.32 specifies the precision and I borrowed this from the tcpdump code that handles dropping root privileges.